app/src/main/java/com/iic/shopingo/ui/request_flow/activities/RequestStateActivity.java

```diff
@@ -40,19 +50,37 @@
   };
 
   private OutgoingRequest request;
+  private boolean firstStart = true;
 
   @Optional
   @InjectView(R.id.request_state_status_explanation)
   TextView statusExplanation;
 
   @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    request = getIntent().getParcelableExtra(EXTRAS_REQUEST_KEY);
-    setContentView(LAYOUTS[request.getStatus().ordinal()]);
-    ButterKnife.inject(this);
-    String explanationString = getString(EXPLANATION_STRINGS[request.getStatus().ordinal()]);
-    statusExplanation.setText(String.format(explanationString, request.getShopper().getFirstName()));
+
+  protected void onStart() {
+    super.onStart();
+    if (firstStart && getIntent().hasExtra(EXTRAS_REQUEST_KEY)) {
+      setRequest((OutgoingRequest)getIntent().getParcelableExtra(EXTRAS_REQUEST_KEY));
+    } else {
+      setRequest(new OutgoingRequestStorage(PreferenceManager.getDefaultSharedPreferences(this)).load());
```

Comment:
I think it really depends on how we end up doing the notifications. Dunno at the moment.